### PR TITLE
Jobs should be fetched and scheduled based upon the selected Shop(#2r65b1c)

### PR DESCRIPTION
**Pull request by disha1202**
### Related Issues

Closes #

### Short Description and Why It's Useful

1. Updated the logic to fetch pending jobs whose shopId is either the selected shop or empty
2. Used ShopifyConfigAndShopLocation (View Entity) instead of ShopifyConfig for fetching shopifyConfig and shopId on login and productStore change
3. Updated the state to store shopifyConfig instead of shopifyConfigId as we're storing shopId as well
4. Passed shopId while scheduling service

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/store/modules/user/actions.ts [hunk 1]
```diff
@@ -136,37 +136,40 @@ const actions: ActionTree<UserState, RootState> = {
...
-        "fieldList": ["shopifyConfigId", "shopifyConfigName"]
+        "fieldList": ["shopifyConfigId", "shopifyConfigName", "shopId"]
       }
       try {
         resp = await UserService.getShopifyConfig(payload);
-        if (resp.status === 200 && !hasError(resp) && resp.data?.docs) {
-          const shopifyConfigs = resp.data.docs;
-          commit(types.USER_SHOPIFY_CONFIG_LIST_UPDATED, shopifyConfigs);
-          commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, shopifyConfigs.length > 0 ? shopifyConfigs[0].shopifyConfigId : "");
+        if (resp.status === 200 && !hasError(resp) && resp.data?.docs.length > 0) {
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:145]: I think we should null for docs as well

File: src/store/modules/user/index.ts [hunk 2]
```diff
@@ -11,8 +11,8 @@ const userModule: Module<UserState, RootState> = {
       token: '',
       current: null,
       instanceUrl: '',
-      shopifyConfigs: [],
-      currentShopifyConfigId: "",
+      shopifyConfigs: {},
```
Comment (adityasharma7) [inline src/store/modules/user/index.ts:14]: I think this should be array only

File: src/store/modules/user/mutation-types.ts [hunk 3]
```diff
@@ -3,6 +3,6 @@ export const USER_TOKEN_CHANGED = SN_USER + '/TOKEN_CHANGED'
 export const USER_END_SESSION = SN_USER + '/END_SESSION'
 export const USER_INFO_UPDATED = SN_USER + '/INFO_UPDATED'
 export const USER_INSTANCE_URL_UPDATED = SN_USER + '/INSTANCE_URL_UPDATED'
-export const USER_SHOPIFY_CONFIG_LIST_UPDATED = SN_USER + '/SHOPIFY_CONFIG_LIST_UPDATED'
+export const USER_SHOPIFY_CONFIGS_UPDATED = SN_USER + '/SHOPIFY_CONFIG_LIST_UPDATED'
```
Comment (adityasharma7) [inline src/store/modules/user/mutation-types.ts:6]:
```suggestion
export const USER_SHOPIFY_CONFIGS_UPDATED = SN_USER + '/SHOPIFY_CONFIGS_UPDATED'
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts [hunk 4]
```diff
@@ -412,10 +418,14 @@ const actions: ActionTree<JobState, RootState> = {
         'runAsUser': 'system', //default system, but empty in run now.  TODO Need to remove this as we are using SERVICE_RUN_AS_SYSTEM, currently kept it for backward compatibility
         'recurrenceTimeZone': this.state.user.current.userTimeZone
       },
-      'shopifyConfigId': this.state.user.currentShopifyConfigId,
       'statusId': "SERVICE_PENDING",
       'systemJobEnumId': job.systemJobEnumId
     } as any
+    
+    if(job?.runtimeData?.shopifyConfigId) {
+      payload['shopifyConfigId'] = this.state.user.currentShopifyConfig?.shopifyConfigId
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:426]: Store `this.state.user.currentShopifyConfig` in a variable and check for it's value in `if` condition itself and remove optional chaining.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts [hunk 5]
```diff
@@ -529,13 +540,13 @@ const actions: ActionTree<JobState, RootState> = {
         'systemJobEnumId': job.systemJobEnumId,
         'tempExprId': job.jobStatus, // Need to remove this as we are passing frequency in SERVICE_TEMP_EXPR, currently kept it for backward compatibility
         'parentJobId': job.parentJobId,
-        'recurrenceTimeZone': this.state.user.current.userTimeZone
+        'recurrenceTimeZone': this.state.user.current.userTimeZone,
+        'shopId': job.status === "SERVICE_PENDING" ? job.shopId : this.state.user.currentShopifyConfig.shopId,
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:544]:
```suggestion
        'shopId': job.runtimeData?.shopifyConfigId && job.status === "SERVICE_PENDING" ? job.shopId : this.state.user.currentShopifyConfig.shopId,
```

File: src/store/modules/job/actions.ts [hunk 6]
```diff
@@ -529,13 +540,13 @@ const actions: ActionTree<JobState, RootState> = {
...
       },
-      'shopifyConfigId': job.status === "SERVICE_PENDING" ? job.runtimeData?.shopifyConfigId : this.state.user.currentShopifyConfigId,
+      'shopifyConfigId': job.status === "SERVICE_PENDING" ? job.runtimeData?.shopifyConfigId : this.state.user.currentShopifyConfig.shopifyConfigId,
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:546]:
```suggestion
      'shopifyConfigId': job.runtimeData?.shopifyConfigId && job.status === "SERVICE_PENDING" ? job.runtimeData?.shopifyConfigId : this.state.user.currentShopifyConfig.shopifyConfigId,
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts (hunk 5 above)
Comment (disha1202) [reply src/store/modules/job/actions.ts]: Updated the condition.

File: src/store/modules/job/actions.ts (hunk 6 above)
Comment (disha1202) [reply src/store/modules/job/actions.ts]: Updated the condition.

**Review by adityasharma7 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**